runtime/js/40_http.js

```diff
@@ -67,8 +67,9 @@
       const request = fromInnerRequest(innerRequest, "immutable");
 
       const respondWith = createRespondWith(responseSenderRid, this.#rid);
+      const upgradeWebsocket = createUpgradeWebsocket(requestBodyRid);
 
-      return { request, respondWith };
+      return { request, respondWith, upgradeWebsocket };
```

Comment:
can we use `upgrade` instaed of `upgradeWebsocket ` ?

Reply:
No, upgrading isn't something exclusive to websockets. that would end up being confusing.

Reply:
> Of course that would make sense if we would support upgrading other kind of conns out of the box

Other hypothetical upgrades would have different return types, so they should just have individual names anyway.